packages/react/src/AppShell/AppShell.tsx

```diff
@@ -23,16 +23,26 @@ export interface AppShellProps {
 
 export function AppShell(props: AppShellProps): JSX.Element {
   const [navbarOpen, setNavbarOpen] = useState(localStorage['navbarOpen'] === 'true');
+  const [isNotificationShown, setIsNotificationShown] = useState(false);
   const medplum = useMedplum();
   const profile = useMedplumProfile();
 
   useEffect(() => {
     function eventListener(): void {
-      showNotification({ color: 'red', message: 'No connection to server', autoClose: false });
+      if (!isNotificationShown) {
+        showNotification({
```

Comment:
I think in this case, we can simply add an `id` property to the `showNotification` argument, and then the Mantine notification system will automatically handle the de-duplication.

That has the added benefit of (in the future) giving the notification a handle that we can use to update status to "back online".  We don't have an event for "online" right now, so that is something in the future.

```suggestion
        showNotification({ id: 'offline', color: 'red', message: 'No connection to server', autoClose: false });
```